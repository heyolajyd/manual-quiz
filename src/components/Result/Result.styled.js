import styled from 'styled-components';

export const ResultContainer = styled.section`
  display: flex;
  flex-direction: column;
  margin-top: 20px;
`;

export const Text = styled.span`
  color: ${(props) => props.theme.colors.grey};
  margin-bottom: 20px;

  a {
    text-decoration: none;
    margin-left: 2px;
  }
`;
