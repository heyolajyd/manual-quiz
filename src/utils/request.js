/**
 * Util function to get questions
 */
export const getQuestion = async () => {
  try {
    const res = await fetch('./questions.json');
    return await res.json();
  } catch (err) {
    console.log('Failed to parse: ', JSON.stringify(err));
    throw new Error(err);
  }
};
