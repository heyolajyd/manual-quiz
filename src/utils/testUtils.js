import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/global';

export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
