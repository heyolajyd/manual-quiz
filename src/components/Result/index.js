import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import { ResultContainer, Text } from './Result.styled';

const Result = ({ startQuiz, isSupportAvailable }) => {
  return (
    <ResultContainer>
      {isSupportAvailable ? (
        <Text>
          Great news! We have the perfect treatment for your hair loss. Proceed
          to
          <a
            href='https://www.manual.co/'
            target='_blank'
            rel='noopener noreferrer'
          >
            www.manual.co
          </a>
          , and prepare to say hello to your new hair!
        </Text>
      ) : (
        <Text>
          Unfortunately, we are unable to prescribe this medication for you.
          This is because finasteride can alter the PSA levels, which maybe used
          to monitor for cancer. You should discuss this further with your GP or
          specialist if you would still like this medication.
        </Text>
      )}
      <Button onClick={startQuiz}>Start Again</Button>
    </ResultContainer>
  );
};

Result.propTypes = {
  isSupportAvailable: PropTypes.bool,
  startQuiz: PropTypes.func,
};

export default Result;
