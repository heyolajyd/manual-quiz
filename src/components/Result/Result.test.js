import React from 'react';
import { fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import { withThemeProvider } from 'utils/testUtils';
import Result from './index';

const startkQuizMock = jest.fn();

describe('Result', function () {
  beforeEach(() => {
    this.props = {
      isSupportAvailable: true,
      startQuiz: startkQuizMock,
    };
  });

  it('should match snapshot', () => {
    const { container } = withThemeProvider(<Result {...this.props} />);
    expect(container).toMatchSnapshot();
  });

  it('should show not supporting text when support if not available', () => {
    this.props.isSupportAvailable = false;
    const { getByText } = withThemeProvider(<Result {...this.props} />);

    expect(getByText(/Unfortunately, we are unable/)).toBeInTheDocument();
  });

  it('should start quiz again', () => {
    const { getByText } = withThemeProvider(<Result {...this.props} />);

    fireEvent.click(getByText(/Start Again/i));

    expect(startkQuizMock).toBeCalled();
  });
});
