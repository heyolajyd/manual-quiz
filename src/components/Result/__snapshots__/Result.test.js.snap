// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should match snapshot 1`] = `
.c2 {
  background-color: #34d1d1;
  cursor: pointer;
  outline: none;
  border-radius: 24px;
  height: 48px;
  padding: 10px 35px;
  color: #fff;
  font-size: 12px;
  font-family: Montserrat;
  font-weight: 700;
  -webkit-letter-spacing: 2.4px;
  -moz-letter-spacing: 2.4px;
  -ms-letter-spacing: 2.4px;
  letter-spacing: 2.4px;
  border: 0;
  max-width: 220px;
  box-shadow: 0 10px 30px -20px rgba(0,0,0,0.5);
}

.c2:disabled {
  color: rgba(16,16,16,0.3);
  background-color: rgba(239,239,239,0.3);
  border-color: rgba(195,195,195,0.3);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 20px;
}

.c1 {
  color: #494949;
  margin-bottom: 20px;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-left: 2px;
}

<div>
  <section
    class="c0"
  >
    <span
      class="c1"
    >
      Great news! We have the perfect treatment for your hair loss. Proceed to
      <a
        href="https://www.manual.co/"
        rel="noopener noreferrer"
        target="_blank"
      >
        www.manual.co
      </a>
      , and prepare to say hello to your new hair!
    </span>
    <button
      class="c2"
    >
      Start Again
    </button>
  </section>
</div>
`;
